backend/audit/fixtures/workbooks/federal_awards.py

```diff
@@ -0,0 +1,271 @@
+    # Map things with transformations
+    prefixes = map(lambda v: (v.cfda).split(".")[0], cfdas)
+    # prefixes = map(lambda v: f'0{v}' if int(v) < 10 else v, prefixes)
+    # Truncate any nastiness in the CFDA extensions to three characters.
+    extensions = map(lambda v: ((v.cfda).split(".")[1])[:3].upper(), cfdas)
+    extensions = map(
+        lambda v: v
+        if re.search("^(RD|[0-9]{3}[A-Za-z]{0,1}|U[0-9]{2})$", v)
+        else "000",
+        extensions,
+    )
+    return (prefixes, extensions)
+
+
+def _fix_passthroughs(cfdas, dbkey):
+    passthrough_names = ["" for x in list(range(0, len(cfdas)))]
+    passthrough_ids = ["" for x in list(range(0, len(cfdas)))]
+    ls = Cfda.select().where(Cfda.dbkey == dbkey).order_by(Cfda.index)
+    cfda: Cfda
+    for cfda in ls:
+        pnq = Passthrough()
+        if cfda.direct == "Y":
+            pnq.passthroughname = ""
+            pnq.passthroughid = ""
+        if cfda.direct == "N":
+            try:
+                pnq = (
+                    Passthrough.select().where(
+                        (Passthrough.dbkey == cfda.dbkey)
+                        & (Passthrough.elecauditsid == cfda.elecauditsid)
+                    )
+                ).get()
+            except Exception as e:
+                print(e)
+                pnq.passthroughname = "EXCEPTIONAL PASSTHROUGH NAME"
+                pnq.passthroughid = "EXCEPTIONAL PASSTHROUGH ID"
+
+        name = pnq.passthroughname
+        if name is None:
+            name = ""
+        name = name.rstrip()
+        if name == "" and cfda.direct == "N":
+            passthrough_names[
+                get_list_index(cfdas, cfda.index)
+            ] = "NO PASSTHROUGH NAME PROVIDED"
+        else:
+            passthrough_names[get_list_index(cfdas, cfda.index)] = name
+
+        id = pnq.passthroughid
+        if id is None:
+            id = ""
+        id = id.rstrip()
+        if id == "" and cfda.direct == "N":
+            passthrough_ids[
+                get_list_index(cfdas, cfda.index)
+            ] = "NO PASSTHROUGH ID PROVIDED"
+        else:
+            passthrough_ids[get_list_index(cfdas, cfda.index)] = pnq.passthroughid
+
+    return (passthrough_names, passthrough_ids)
+
+
+def generate_federal_awards(dbkey, outfile):
+    logger.info(f"--- generate federal awards {dbkey}---")
+    wb = pyxl.load_workbook(templates["FederalAwards"])
+    # In sheet : in DB
+
+    g = set_uei(Gen, wb, dbkey)
+    insert_version_and_sheet_name(wb, "federal-awards-workbook")
+
+    cfdas = Cfda.select().where(Cfda.dbkey == dbkey).order_by(Cfda.index)
+    map_simple_columns(wb, mappings, cfdas)
+
+    # Patch the clusternames. They used to be allowed to enter anything
+    # they wanted.
+    valid_file = open(f"{settings.BASE_DIR}/schemas/source/base/ClusterNames.json")
+    valid_json = json.load(valid_file)
+    # This was removed from the CSV...
+    valid_json["cluster_names"].append("STATE CLUSTER")
```

Comment:
We removed N/A as well. Do we need something like this? 

```suggestion
    valid_json["cluster_names"].append("STATE CLUSTER")
    valid_json["cluster_names"].append("N/A")
```